### ISSUE TITLE
.pth files generated by buildPythonPackage do not include transitive dependencies

### ISSUE THREAD

**Issue by adnelson** [NixOS/nixpkgs#7816]
Let's say I have an expression like so:

```nix
with import <nixpkgs> {};
rec {
  foo = buildPythonPackage {
    name = "foo";
    src = /path/to/foo;
  };
  bar = buildPythonPackage {
    name = "bar";
    src = /path/to/bar;
    propagatedBuildInputs = [foo];
  };
  baz = buildPythonPackage {
    name = "baz";
    src = /path/to/baz;
    propagatedBuildInputs = [bar];
  };
}
```

Now, if I do `nix-env -iA bar`, then of course it will first build foo and then bar, and in my `~/.nix-profile/lib/python2.7/site-packages` folder, I will see a `bar-propagated-inputs.pth` (more or less) containing something like `/nix/store/23984vhdbji34589unsdo67234y-python2.7-foo/lib/python2.7/site-packages`.

But now let's say that without installing `bar`, I instead did `nix-env -iA baz`. It will properly build `foo` and `bar`, but now in `~/.nix-profile/lib/python2.7/site-packages` I will see a `baz-propagated-inputs.pth` file, which references `/nix/store/23o48hdfio87245h96unr6-python2.7-bar/lib/python2.7/site-packages`, but **does not** reference the path to `foo`'s site-packages. This means that, if one is using python's `site` package to automagically load paths, only the paths for `bar` and `baz` will appear in `sys.path`, and not `foo`, which almost certainly will mean that using either of those modules will result in a runtime error.

**TL;DR:** Either: 
  * the `.pth` file generated by a python package should contain not only its own dependencies, but those of its dependencies as well, or
  * Doing an installation of a python package should add to the nix profile not only that package's `.pth` file, but those of its dependencies as well.

**Comment by domenkozar** [comment]
`.pth` files are now gone in #11086

**Comment by nbp** [comment]
@domenkozar I guess we should close this issue?

**Comment by domenkozar** [comment]
@adnelson I recommend usage of `nix-shell` or `python.buildEnv`. Let me know if you have some questions
